core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -512,6 +528,50 @@ private ExprValue exprHour(ExprValue time) {
     return new ExprIntegerValue(time.timeValue().getHour());
   }
 
+  /**
+   * Following MySQL, function receives arguments of type double and rounds them before use.
+   * Furthermore:
+   *  - zero year interpreted as 2000
+   *  - negative year is not accepted
+   *  - @dayOfYear should be greater than 1
+   *  - if @dayOfYear is greater than 365/366, calculation goes to the next year(s)
+   *
+   * @param yearExpr year
+   * @param dayOfYearExp day of the @year, starting from 1
+   * @return Date - ExprDateValue object with LocalDate
+   */
+  private ExprValue exprMakeDate(ExprValue yearExpr, ExprValue dayOfYearExp) {
+    var year = Math.round(yearExpr.doubleValue());
+    var dayOfYear = Math.round(dayOfYearExp.doubleValue());
+    // We need to do this to comply with MySQL
+    if (0 >= dayOfYear || 0 > year) {
```

Comment:
nit: Can we make this check prior to rounding?  Then if the value is negative, we save those operations.

Reply:
To optimize performance we shouldn't use `LocalDate::parse` nor `LocalDate::plusDays`. See my note about faster and easier implementation.